anitya/lib/backends/npmjs.py

```diff
@@ -92,3 +92,28 @@ def get_versions(cls, project):
             raise AnityaPluginException('No versions found at %s' % url)
 
         return data['versions'].keys()
+
+    @classmethod
+    def check_feed(cls):
+        ''' Return a generator over the latest 40 uploads to npmjs.org
+
+        by querying an weird JSON endpoint.
+        '''
+
+        url = 'https://registry.npmjs.org/-/all/static/today.json'
+
+        try:
+            response = cls.call_url(url)
+        except Exception:  # pragma: no cover
+            raise AnityaPluginException('Could not contact %s' % url)
+
+        try:
+            data = response.json()
+        except Exception:  # pragma: no cover
+            raise AnityaPluginException('No JSON returned by %s' % url)
+
+        for item in data[:40]:
```

Comment:
> the record for that package is never created in Anitya and hence it won't be discovered until its new release is caught by this code again.

This is correct.

- @pypingou, our current "full scan" mode scans for all of the projects that anitya already knows about, and it does so very well.
- This new mode not only updates projects that anitya already know about, but it also creates entries for *new* projects that it discovers.

If we worry only about the first part here, then it is ok to miss an announcement due to the aliasing effect at play here, since our full scan will pick up the release that we missed.  However, like @voxik says, we run the risk of failing to discover new projects if they get pushed off the list between our 5-minute scan windows.